src/crl.c

```diff
@@ -600,10 +604,63 @@ static void SetCrlInfoFromDecoded(DecodedCRL* entry, CrlInfo *info)
     info->nextDate = (byte *)entry->nextDate;
     info->nextDateMaxLen = MAX_DATE_SIZE;
     info->nextDateFormat = entry->nextDateFormat;
-    info->crlNumber = (sword32)entry->crlNumber;
+    XMEMCPY(info->crlNumber, entry->crlNumber, CRL_MAX_NUM_SZ);
+    info->crlNumberSet = entry->crlNumberSet;
 }
 #endif
 
+/* Returns 1 if prev crlNumber is smaller
+ *         0 if equal
+ *        -1 if prev crlNumber is larger */
+static int CompareCRLnumber(CRL_Entry* prev, CRL_Entry* curr)
+{
+    int result = 0;
+#ifdef WOLFSSL_SMALL_STACK
+    mp_int* prev_num = (mp_int*)XMALLOC(sizeof(*prev_num), NULL,
```

Comment:
Regarding @SparkiDev 's concern about allocing gobs of memory just to process 20 octets of MPI:  It's possible to right-size-allocate the MPI containers with existing macros.  You'd use `DECL_MP_INT_SIZE()`, `NEW_MP_INT_SIZE()`. `INIT_MP_INT_SIZE()`, and `FREE_MP_INT_SIZE()`.  These exist with all MPI back ends, though they only right-size on SP and heapmath, not TFM (doesn't matter).

Reply:
Refactored to new function to use these macros :)